spring-boot-project/spring-boot/src/main/java/org/springframework/boot/env/OriginTrackedYamlLoader.java

```diff
@@ -165,16 +165,39 @@ private static Node get(Node node) {
 	}
 
 	/**
-	 * {@link Resolver} that limits {@link Tag#TIMESTAMP} tags.
+	 * {@link Resolver} that limits {@link Tag#TIMESTAMP} tags, keeps only true and false
+	 * as booleans, drops octal integers, base 2 integers, NaN numbers.
 	 */
 	private static class NoTimestampResolver extends Resolver {
 
+		public static final Pattern SPRING_BOOL = Pattern.compile("^(?:true|True|TRUE|false|False|FALSE)$");
+
+		public static final Pattern SPRING_FLOAT = Pattern
+				.compile("^(" + "[-+]?(?:[0-9][0-9_]*)\\.[0-9_]*(?:[eE][-+]?[0-9]+)?" + // (base
+																						// 10)
+						"|[-+]?(?:[0-9][0-9_]*)(?:[eE][-+]?[0-9]+)" + // (base 10,
+																		// scientific
+																		// notation
+																		// without .)
+						"|[-+]?\\.[0-9_]+(?:[eE][-+]?[0-9]+)?" + // (base 10, starting
+																	// with .)
+						"|[-+]?\\.(?:inf|Inf|INF)" + ")$");
+
+		public static final Pattern SPRING_INT = Pattern.compile("^(?:" + "|[-+]?(?:0|[1-9][0-9_]*)" + // (base
```

Comment:
Shouldn't this (and hex below) be limited in length?

Reply:
The spec does not put any restriction. The restriction comes from CVE which try to send invalid data and report an issue.